### PR TITLE
Validate snapshot lifecycle policies

### PR DESCRIPTION
**Pull request by dakrone**
(This is a PR against the `snapshot-lifecycle-management` branch)

With the commit, we now validate the content of snapshot lifecycle policies when
the policy is being created or updated. This checks for the validity of the id,
name, schedule, and repository. Additionally, cluster state is checked to ensure
that the repository exists prior to the lifecycle being added to the cluster
state.

Part of #38461

### CODE REVIEW

**Review by AthenaEryma — CHANGES_REQUESTED** [review]
Left some minor comments, otherwise looks good.

File: x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/snapshotlifecycle/SnapshotLifecycleService.java [hunk 1]
```diff
@@ -176,6 +178,16 @@ public void cancelScheduledSnapshot(final String lifecycleJobId) {
         scheduler.remove(lifecycleJobId);
     }
 
+    /**
+     * Validates that the {@code repository} exists as a registered snapshot repository
+     * @throws IllegalArgumentException if the repository does not exist
+     */
+    public static void validateRepositoryExists(final String repository, final ClusterState state) {
```
[the remark this answers is not in the document]
Comment (dakrone) [reply x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/snapshotlifecycle/SnapshotLifecycleService.java]: I was thinking about that, I should be able to change it to a valid repo, but list `"indices": ["foo"], "ignore_unavailable": false` and then it should fail if the "foo" index is not available.

[the remark this answers is not in the document]
Comment (gwbrown) [reply x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/snapshotlifecycle/SnapshotLifecycleService.java]: Oh, cool. I'll make that change in my PR before it gets merged.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/snapshotlifecycle/SnapshotLifecyclePolicy.java [hunk 2]
```diff
@@ -108,9 +113,64 @@ public String getRepository() {
         return this.configuration;
     }
 
-    public ValidationException validate() {
-        // TODO: implement validation
-        return null;
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException err = new ActionRequestValidationException();
+
+        // ID validation
```
[the remark this answers is not in the document]
Comment (dakrone) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/snapshotlifecycle/SnapshotLifecyclePolicy.java]: Yeah not a bad idea, I copied this from ILM's validation for now.

**Comment by dakrone** [comment]
Thanks for taking a look Gordon, I pushed commits addressing your comments

**Review by AthenaEryma — APPROVED** [review]
LGTM!

**Pull request merged**